larsim/LegacyLArG4/OpFastScintillation.cxx

```diff
@@ -1550,6 +1548,10 @@ namespace larg4 {
     for (size_t const OpDet : util::counter(NOpChannels)) {
       if (!isOpDetInSameTPC(ScintPoint, fOpDetCenter.at(OpDet))) continue;
 
+      // set detector struct for solid angle function
+      detPoint.h = fOpDetHeight.at(OpDet);
+      detPoint.w =  fOpDetLength.at(OpDet);
```

Comment:
This seems to be an appropriate change.  However, instead of `detPoint` being a class data member, I would move it to be a stack variable that is passed in to the relevant functions (e.g.):
```.cpp
dims const det_point{fOpDetHeight.at(OpDet), fOpDetLength.at(OpDet)};
VUVHits(Num, ScintPoint, fOpDetCenter[OpDet], fOpDetType[OpDet], det_point);
```
I understand this changes the internal interface a little bit, but I think it is worth it.

Reply:
I'll take your advice on this, I'll make the changes in a bit.